### ISSUE TITLE
Crashes when duplicate files are scanned

### ISSUE THREAD

**Issue by WFlyToTheSky** [gorilla-devs/ferium#422]
### Description

I don't know if this is a bug but i cant figure out why this is happening it just gives this error when i run ferium scan:
```
Querying Servers:
thread 'main' panicked at /home/wally/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libium-1.29.0/src/scan.rs:74:22:
Missing filename in hashmap
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### Steps to reproduce

1. Run ferium profile create
2. set the version to 1.20.1
3. set the modloader to fabric
4. set the path to /home/myname/Desktop/FabricServer/mods
5. run ferium scan

### Operating System

macOS

### Are you using the latest version of ferium?

- [X] I am using the latest version of ferium

**Comment by theRookieCoder** [comment]
I hope you don't mind, but it would be much easier for me to debug this if I can see what files you have in that folder you're scanning. You can contact me on Discord (`therookiecoder`) so we can discuss this privately.

**Comment by theRookieCoder** [comment]
Fixed in `v4.7.1`
